scripts/sync-pkgs-wireit.js

```diff
@@ -0,0 +1,186 @@
+const { log, error } = console; // eslint-disable-line no-undef
+
+// ============================================================================
+// Config
+// ============================================================================
+const ROOT = path.resolve(__dirname, ".."); // eslint-disable-line no-undef
+const PKGS_ROOT = path.join(ROOT, "packages");
+
+// Special packages
+const PKGS = {
+  CORE: "victory-core",
+  NATIVE: "victory-native",
+  VENDOR: "victory-vendor",
+};
+const SPECIAL_PKGS = new Set([PKGS.NATIVE, PKGS.VENDOR]);
+
+// ============================================================================
+// Helpers
+// ============================================================================
+const readPkg = async (pkgPath) => JSON.parse(await fs.readFile(pkgPath));
+const writePkg = async (pkgPath, data, originalPkg) => {
+  const json = JSON.stringify(data, null, 2);
+  if (json === JSON.stringify(originalPkg, null, 2)) {
+    log(`Skipping ${pkgPath} (no changes)`);
+    return;
+  }
+  log(`Writing ${pkgPath}`);
+  await fs.writeFile(pkgPath, `${json}\n`);
+};
+const clone = (obj) => JSON.parse(JSON.stringify(obj));
+const isVictoryPackage = (p) => p.startsWith("victory");
+
+// Root mutation
+//
+// We want to use wireit directly to manage multi-build for better
+// cache hits (e.g. `pnpm -r run build` seems to get a lot of cache
+// misses). So create tasks with cross-package deps
+const updateRootPkg = async ({ allPkgs }) => {
+  const rootPkgPath = `${ROOT}/package.json`;
+  const originalPkg = await readPkg(rootPkgPath);
+  const rootPkg = clone(originalPkg);
+
+  rootPkg.wireit = rootPkg.wireit || {};
+  [
+    { rootTask: "build", pkgTask: "build" },
+    { rootTask: "format:pkgs", pkgTask: "format" },
+    { rootTask: "lint:pkgs", pkgTask: "lint" },
+    { rootTask: "jest:pkgs", pkgTask: "jest" },
+    { rootTask: "types:check:pkgs", pkgTask: "types:check" },
+    { rootTask: "types:create:pkgs", pkgTask: "types:create" },
+  ].forEach(({ rootTask, pkgTask }) => {
+    rootPkg.wireit[rootTask] = rootPkg.wireit[rootTask] || {};
+    rootPkg.wireit[rootTask].dependencies = allPkgs.map(
+      (p) => `./packages/${p}:${pkgTask}`,
+    );
+  });
+
+  await writePkg(rootPkgPath, rootPkg, originalPkg);
+};
+
+// Common library mutations.
+//
+// Use the core package as the template for the rest.
+const updateLibPkgs = async ({ libPkgs }) => {
+  const corePkg = await readPkg(`${PKGS_ROOT}/victory-core/package.json`);
+  const rootPkg = await readPkg(`${ROOT}/package.json`);
+
+  for (const workspace of libPkgs) {
+    const pkgPath = `${PKGS_ROOT}/${workspace}/package.json`;
+    const originalPkg = await readPkg(pkgPath);
+    const pkg = clone(originalPkg);
+
+    // Overwrite scripts and wireit configuration
+    pkg.scripts = clone(corePkg.scripts);
+    pkg.wireit = clone(corePkg.wireit);
+
+    // Clear out existing deps from victory-core
+    // TODO(wireit): Abstract and refactor this whole section better.
```

Comment:
I agree with this comment, and have an idea...
Currently, we're pulling in our "source config" from `victory-core/package.json`, and then augmenting it here.

Instead, let's actually have the whole "source config" here in this file.  Here's a sample of what I'm thinking:

```js
function generateWireitConfig(pkg, rootPkg) {
  const isVictoryPackage = (p) => p.startsWith("victory") && p !== pkg.name; 
  const deps = Object.keys(pkg.dependencies).filter(isVictoryPackage);
  const devDeps = Object.keys(pkg.devDependencies).filter(isVictoryPackage);
  const rootDeps = Object.keys(rootPkg.devDependencies).filter(isVictoryPackage);
  return {
    "scripts": {
      "build": "wireit",
      "build:lib:esm": "wireit",
      // ...
      "types:create": "wireit",
      "format": "wireit",
      "lint": "wireit",
      "jest": "wireit"
    },
    "wireit": {
      "build": {
        "dependencies": [
          "build:lib",
          "build:dist",
          "types:create"
        ]
      },
      "build:lib:esm": {
        "command": "nps build:lib:esm",
        "files": [
          "src/**",
          "!src/**/*.test.*",
          "../../.babelrc.build.js"
        ],
        "output": [
          "es/**/*.js",
          "es/**/*.js.map"
        ],
        "dependencies": [
          ...deps.map(dep => `../${dep}:build:lib:esm`),
        ]
      },
      // ...
      "types:check": {
        "command": "nps types:pkg:check",
        "files": [
          "src/**/*.{ts,tsx}",
          "../../tsconfig.base.json",
          "tsconfig.json"
        ],
        "dependencies": [
          "types:create",
          ...[...deps, ...devDeps, ...rootDeps].map(dep => `../${dep}:types:create`),
        ],
        "output": []
      },
      "types:create": {
        "command": "nps types:pkg:create",
        "files": [
          "src/**",
          "!src/**/*.test.*",
          "../../tsconfig.base.json",
          "tsconfig.build.json"
        ],
        "output": [
          "es/**/*.d.ts",
          "es/**/*.d.ts.map",
          "lib/**/*.d.ts",
          "lib/**/*.d.ts.map"
        ],
        "dependencies": [
          ...deps.map(dep => `../${dep}:types:create`),
        ]
      },
      "jest": {
        "command": "nps jest:pkg",
        "files": [
          "src/**/*.test.*",
          "../../.babelrc.js",
          "../../jest-config.js",
          "../../test/jest-setup.js"
        ],
        "output": [],
        "dependencies": [
          "build:lib:cjs",
          ...[...deps, ...devDeps, ...rootDeps].map(dep => `../${dep}:build:lib:cjs`),
        ]
      }
    }
  }
}


```

Reply:
(Merged)